--- src/code/print.lisp.old     2009-04-24 16:41:55.000000000 +0200
+++ src/code/print.lisp 2009-11-18 14:14:38.000000000 +0100
@@ -1547,7 +1547,7 @@
   (let ((*print-radix* nil))
     (if (typep x *read-default-float-format*)
         (unless (eql exp 0)
-          (format stream "e~D" exp))
+          (format stream "E~D" exp))
         (format stream "~C~D"
                 (etypecase x
                   (single-float #\f)
@@ -1622,6 +1622,8 @@
       (t (write-string string stream :end 1)
          (write-char #\. stream)
          (write-string string stream :start 1)
+        (when (<= (length string) 1)    
+               (write-char #\0 stream))
          (print-float-exponent x (1- e) stream)))))
 
 ;;;; other leaf objects
